app/gui/src/project-view/providers/widgetRegistry.ts

```diff
@@ -374,7 +373,7 @@ export function defineWidget<M extends InputMatcher<any> | InputMatcher<any>[]>(
     widgetTypeId: crypto.randomUUID() as WidgetTypeId,
   }
 
-  if (import.meta.hot && hmr) {
+  if (import.meta.hot && hmr && hmr.data) {
```

Comment:
Why is this check necessary? `hmr.data` should always be a defined object that you are supposed to assign properties to.

Docs: https://vite.dev/guide/api-hmr#hot-data